lib/global_uid/base.rb

```diff
@@ -179,6 +200,7 @@ def self.get_many_uids_for_class(klass, count, options = {})
       with_connections do |connection|
         Timeout.timeout(self.global_uid_options[:query_timeout], TimeoutException) do
           increment_by = connection.select_value("SELECT @@auto_increment_increment")
+          raise InvalidIncrementPreventionException if increment_by != self.global_uid_options[:increment_by]
```

Comment:
It's a little odd to track `server[:allocated_ids]` for single allocations but leave it completely out for multiple allocations. If someone were trying to troubleshoot id generation using `server[:allocated_ids]`, they could be getting misleading information.

Reply:
We don't need to track them here as we've just queried the database for the value of `auto_increment_increment ` and can compare it with what the client has configured `global_uid_options[:increment_by]`, raising an exception before we create any records.